### PR TITLE
Program Boilerplate update for Billing/Feature enforcement

### PR DESCRIPTION
**Pull request by jayden-chan**
## Description of the change

> Description here

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

- Jira issue number: (PUT IT HERE)
- Process.st launch checklist: (PUT IT HERE)

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run (if applicable)
- [x] The behaviour changes in the pull request are covered by specs
- [x] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist

### Code review

- [x] Changes have been reviewed by at least one other engineer
- [x] Security impacts of this change have been considered

### CODE REVIEW

File: packages/program-boilerplate/src/trigger.ts
Comment (johanventer) [inline packages/program-boilerplate/src/trigger.ts:132]: Is the other case where it was just calling `handleIntrospection(template, rules)` no longer valid?

**Comment by johanventer** [comment]
Needs version bump before approval

File: packages/program-boilerplate/src/trigger.ts
Comment (jayden-chan) [reply packages/program-boilerplate/src/trigger.ts:132]: Hmm I guess this could technically be a breaking change. I removed that because the function signature was changed to have the program and tenant info non-optional (not a breaking change) because that's how the backend schema is written. That means the function can't be called without those arguments so I just deleted the other call to it.

This would only break programs that return `undefined` from introspection when the program/tenant ARE present but return something else if they aren't... I can't imagine that's a valid use case but I guess it could technically be a breaking change. Not sure if it's worth sifting through all the program code to check if any behave like that.

Comment (johanventer) [reply packages/program-boilerplate/src/trigger.ts:132]: Sounds like a strange case, and won't apply to the referral program so I'm ok with this.

**Review by johanventer — APPROVED** [review]

**Pull request merged**
